src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/OnStartupJobLauncherFactory.kt

```diff
@@ -44,6 +45,22 @@ class OnStartupJobLauncherFactory(
       return exitCodeMapper.intValue(execution.exitStatus.exitCode)
     }
 
+    return makeLauncher(job.name, entryPoint)
+  }
+  fun makeReadOnlyBatchLauncher(job: Job, readOnlyJobLauncher: TaskExecutorJobLauncher): ApplicationRunner {
```

Comment:
makeBatchLauncher and makeReadOnlyBatchLauncher almost the same code with the only the launcher is difference. could you please create the readonlylauncher as another bean and as that as a parameter to a private method inside. That private method will be used both by `makeBatchLauncher` and makeReadOnlyBatchLauncher`

Reply:
No worries Kumar, will work that up this morning.

Reply:
Refactored as discussed @kth13 please review